### PR TITLE
Use function instead of commands for text_edit.

### PR DESCRIPTION
**Pull request by hrsh7th**
Change text_edit implementation to use `setbufline`, `appendbufline` and `deletebufline` instead of normal-mode commands.

It is mostly passed test-cases.
But one test case is not passed, I think it is test-case side problem.

### CODE REVIEW

File: test/lsp/utils/text_edit.vimspec [hunk 1]
```diff
@@ -548,7 +548,7 @@ Describe lsp#utils#text_edit
                     \ }])
 
             let l:buffer_text = s:get_text()
-            Assert Equals(l:buffer_text, ['x', 'y', 'z', 'b', ''])
```
Comment (hrsh7th) [inline test/lsp/utils/text_edit.vimspec]: IMO, The buffer is `['foo', 'b']` and range-end is `line: 1, char: 1`.
So `b` character is in range to replacement.

File: autoload/lsp/utils/text_edit.vim [hunk 2]
```diff
@@ -44,42 +17,67 @@ function! lsp#utils#text_edit#apply_text_edits(uri, text_edits) abort
     " ((0, 4), (0, 5), "") - remove fifth character 'e'
     " ((0, 2), (0, 3), "") - remove third character 'c'
     let l:text_edits = sort(deepcopy(a:text_edits), '<SID>sort_text_edit_desc')
-    let l:i = 0
 
+    let l:path = lsp#utils#uri_to_path(a:uri)
+    let l:bufnr = bufnr(l:path, v:true)
+    if !bufloaded(l:path)
+        call bufload(l:path)
```
Comment (hrsh7th) [inline autoload/lsp/utils/text_edit.vim]: Sorry, CI is failed on this line.
`bufload` seems to be a new api...

Comment (hrsh7th) [reply autoload/lsp/utils/text_edit.vim]: I can't found the way to load buffer with no side-effect.
So I will close this PR later...

Comment (thomasfaingnaert) [reply autoload/lsp/utils/text_edit.vim]: @hrsh7th 
What about storing the current buffer in a variable, running `:edit`, and restoring the current buffer again?

Comment (hrsh7th) [reply autoload/lsp/utils/text_edit.vim]: I trying to the solution, thanks. It works well.

https://github.com/prabirshrestha/vim-lsp/compare/master...hrsh7th:change-text-edit-implementation?expand=1

**Pull request closed without merging**